sendbuffer.go

```diff
@@ -66,6 +66,9 @@ func (buf *sendBuffer) sendLoop(w io.Writer) {
 		buf.closing = true
 		close(buf.in)
 		buf.muClosing.Unlock()
+		if !closeTimer.Stop() {
+			<-closeTimer.C
```

Comment:
@oxtoacart do you remember why `signalClose` is running in a new goroutine? It doesn't seem to have block operations.

Per the doc, to reset `closeTimer` correctly, it has to be in the same goroutine of waiting for it.

Reply:
IIRC we ran into a condition where signalClose would block indefinitely if we ran it in the same goroutine as the sendLoop.

Reply:
Oh, and `buf.muClosing.Lock()` is a blocking operation if the mutex is held by something else.